File: src/main/java/com/interviewplannerapp/service/SkillService.java
package com.interviewplannerapp.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.interviewplannerapp.domain.Skill;
import com.interviewplannerapp.dto.SkillDTO;
import com.interviewplannerapp.dto.SkillSearchDTO;
import com.interviewplannerapp.dto.SkillPageDTO;
import com.interviewplannerapp.dto.SkillConvertCriteriaDTO;
import com.interviewplannerapp.service.GenericService;
import com.interviewplannerapp.dto.common.RequestDTO;
import com.interviewplannerapp.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface SkillService extends GenericService<Skill, Integer> {

	List<Skill> findAll();

	ResultDTO addSkill(SkillDTO skillDTO, RequestDTO requestDTO);

	ResultDTO updateSkill(SkillDTO skillDTO, RequestDTO requestDTO);

    Page<Skill> getAllSkills(Pageable pageable);

    Page<Skill> getAllSkills(Specification<Skill> spec, Pageable pageable);

	ResponseEntity<SkillPageDTO> getSkills(SkillSearchDTO skillSearchDTO);
	
	List<SkillDTO> convertSkillsToSkillDTOs(List<Skill> skills, SkillConvertCriteriaDTO convertCriteria);

	SkillDTO getSkillDTOById(Integer skillId);







}

File: interviewplannerapp_ui/src/services/SkillService.js
import http from "../http-common"; 

class SkillService {
  getAllSkills(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/skill/skills`, searchDTO);
  }

  get(skillId) {
    return this.getRequest(`/skill/${skillId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/skill?field=${matchData}`, null);
  }

  addSkill(data) {
    return http.post("/skill/addSkill", data);
  }

  update(data) {
  	return http.post("/skill/updateSkill", data);
  }
  
  uploadImage(data,skillId) {
  	return http.postForm("/skill/uploadImage/"+skillId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new SkillService();
